client/requirements.txt

```diff
@@ -0,0 +1,31 @@
+anyio >= 3.7.1, < 4.0.0
```

Comment:
I still think we should move this up to the root directory as `client-requirements.txt` and de-dupe the `prefect` `requirements*.txt` files.  Maybe in another PR after this one, but let's not lose that thread.

Reply:
I took care of this in this PR - I think it's a great way to make sure that we keep the client+full requirements up to date with each other.